src/main/java/com/gotocompany/depot/message/proto/converter/fields/IntegerProtoField.java

```diff
@@ -0,0 +1,42 @@
+package com.gotocompany.depot.message.proto.converter.fields;
+
+import com.google.protobuf.Descriptors;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+public class IntegerProtoField implements ProtoField {
+    private final Descriptors.FieldDescriptor descriptor;
+    private final Object fieldValue;
+
+    public IntegerProtoField(Descriptors.FieldDescriptor descriptor, Object fieldValue) {
+        this.descriptor = descriptor;
+        this.fieldValue = fieldValue;
+    }
+
+    @Override
+    public Object getValue() {
+        if (fieldValue instanceof Collection<?>) {
+            return ((Collection<?>) fieldValue).stream().map(this::getValue).collect(Collectors.toList());
+        }
+        return getValue(fieldValue);
+    }
+
+    public Long getValue(Object field) {
+        return Long.valueOf(field.toString());
+    }
+
+    @Override
+    public boolean matches() {
+        return descriptor.getType() == Descriptors.FieldDescriptor.Type.INT64
```

Comment:
You can use JavaType instead of using FieldType. That would reduce all these conditions to one condition.

Reply:
yea. i will check

Reply:
this didn't work. some of the unit tests were failing so i reverted it.